dspace-api/src/main/java/org/dspace/content/BitstreamServiceImpl.java

```diff
@@ -403,7 +403,9 @@ public Bitstream getFirstBitstream(Item item, String bundleName) throws SQLExcep
 
     @Override
     public Bitstream getThumbnail(Context context, Bitstream bitstream) throws SQLException {
-        Pattern pattern = Pattern.compile("^" + bitstream.getName() + ".([^.]+)$");
+        Pattern pattern = Pattern.compile("^" +
+            (bitstream.getName() != null ? Pattern.quote(bitstream.getName()) : bitstream.getName())
```

Comment:
I think that could be better if we extract this `Pattern` generation into a proper function, just to be more readable.
Do you agree @floriangantner?

Reply:
@vins01-4science I've extracted the functionality an will re-request the review if the pipeline has passed